recipes/views.py

```diff
@@ -174,7 +180,11 @@ def perform_create(self, serializer):
 
 class RecipeDetailView(RetrieveUpdateDestroyAPIView):
     serializer_class = RecipeSerializer
-    queryset = Recipe.objects.all()
+    queryset = Recipe.objects.all().annotate(
+        average_rate=Coalesce(Avg('rates__score'), Value(0)),
+        like_count=Count('likes', distinct=True),
+        rate_count=Count('rates', distinct=True),
+    )
```

Comment:
Looks great but tests are missing :).

We are using same annotations for two times. In future, you can add a custom manager for `Recipe` model. 
We can use it as `Recipe.objects.all().annotate_average_rate().annotate_like_count().annotate_rate_count()`